### PR TITLE
2pc: TxPreparedPool

### PR DESCRIPTION
**Pull request by sougou**
Infrastructural component for managing prepared connections.

### CODE REVIEW

File: go/vt/tabletserver/tx_prep_pool.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+// is done by TxPool.
+type TxPreparedPool struct {
+	mu    sync.Mutex
+	conns map[string]*preparedConn
+}
+
+type preparedConn struct {
+	*DBConn
+	// inUse will contain a non-empty purpose if the conn is in use.
+	inUse string
```
Comment (alainjobart) [inline go/vt/tabletserver/tx_prep_pool.go]: Why not name this 'purpose' too?

Comment (sougou) [inline go/vt/tabletserver/tx_prep_pool.go]: Done.

**Comment by sougou** [comment]
Still need LGTM.

**Comment by alainjobart** [comment]
LGTM

**Pull request merged**